packages/builder/src/schemas.zod.ts

```diff
@@ -0,0 +1,304 @@
+import { z } from 'zod';
+import { handleZodErrors } from './error/zod';
+
+/// ================================ INPUT CONFIG SCHEMAS ================================ \\\
+
+export const contractSchema = z
+  .object({
+    artifact: z.string(),
+  })
+  .merge(
+    z
+      .object({
+        create2: z.boolean(),
```

Comment:
can we share an example of how we would add documentation description text to these types? I would like to do so myself if we think its out of scope, but it would be great to be able to just encode descriptions for what all of these fields do right in the code!

Reply:
Yes will add this into my next batch of changes.

Reply:
Here's a guide on how to do this with typedoc: 

https://typedoc.org/guides/doccomments/